armi/reactor/assemblies.py

```diff
@@ -206,13 +206,28 @@ def insert(self, index, obj):
 
     def moveTo(self, locator):
         """Move an assembly somewhere else."""
+        oldSymmetryFactor = self.getSymmetryFactor()
         composites.Composite.moveTo(self, locator)
         if self.lastLocationLabel != self.DATABASE:
             self.p.numMoves += 1
             self.p.daysSinceLastMove = 0.0
         self.parent.childrenByLocator[locator] = self
         # symmetry may have changed (either moving on or off of symmetry line)
         self.clearCache()
+        self.scaleParamsToNewSymmetryFactor(oldSymmetryFactor)
+
+    def scaleParamsToNewSymmetryFactor(self, oldSymmetryFactor=1.0):
+        volIntegratedParamsToScale = self.getBlocks()[0].p.paramDefs.atLocation(
+            ParamLocation.VOLUME_INTEGRATED
+        )
+        scalingFactor = oldSymmetryFactor / self.getSymmetryFactor()
+        for b in self.getBlocks():
+            for param in volIntegratedParamsToScale:
```

Comment:
Okay, so this seems suspect.  You are looping over EVERY parameter in a block, and doing a volume integration.  This is obviously done elsewhere in the code.

So, for instance, you have [a failing test](https://github.com/terrapower/armi/actions/runs/12147041096/job/33872380305?pr=2017#step:6:382), and the error is:

```python
    self.assertEqual(params["power"][0], params["power"][1])
    # AssertionError: 360.0940445113585 != 120.03134817045283
```

So, you are applying this scaling factor to the `power` parameter for this block. Thus tripling the power of every block in the reactor, thus tripling the power of the entire reactor.

That is a HUGE change, and from the description on this PR, my guess is that you don't think ARMI miscalculates the power (and every other parameter) by a factor of 3. RIght? Please say yes.

Reply:
Resolved not seeing the error on local, this is apparently the sort of change that requires a reinstall of ARMI.

Reply:
I've located one reason for the failing test, the core model used in the test is from the example notebook, and has a dummy power distribution that does not account for the center assembly only having 1/3rd of itself in the core. In progressing the simulation from c0n0 to c0n1 a fuel shuffle occurs. This shuffle exchanges the center assembly (assembly A) with some other assembly in the core (assembly B). So assembly A would triple in power with this feature, moving from the center location with symmetry factor 3 to somewhere with symmetry factor 1. The opposite would happen to assembly B.

Reply:
The example reactor in https://github.com/terrapower/armi/blob/main/armi/tests/tutorials/data_model.ipynb reports **incorrect** power densities for blocks in the central location. This is because the volume that is generated by the `b.getVolume()` call in the screenshot is reduced by the symmetry factor but the power is not. 
![image](https://github.com/user-attachments/assets/9be3b2a2-8cf1-4e7b-8505-4eecb6b1b463)

Reply:
Additionally, the unit test 
`self.assertEqual(params["power"][0], params["power"][1])`
is incorrect to assert this. From what I can tell, ARMI is built to assume assemblies that are split by boundaries do not contribute their entire power values to the modeled section of the core. The 0 index here is when the assembly is wholly within the modeled core, index 1 is after the assembly is moved to the central location and is only 1/3rd in the modeled core.

Reply:
@keckler @drewj-tp Brian believe the power in the center assembly is off by 1/3. If he's right, this is a one-line fix in ARMI (and it's already done in this PR). But if he's wrong, I certainly don't want to make the change.

Reply:
This is not true, I am not asserting the power is wrong in the center assembly for all reactors. I haven't checked yet, but I would guess that the `power` parameter is handled specially to account for 1/3rd symmetry, and/or during a core solve the power should be calculated correctly. What I am saying is some other parameters are not adjusted to account for the symmetry factor.

That said, the power used in the data model example notebook is wrong (aside from how it was constructed) because when the low power assembly is shuffled into the center it has the same power as it had before the shuffle. This property is then used in a unit test (the one that was discussed as failing above) to assert something that should not be true.

Reply:
I am taking a look at this now...

I think that @bsculac 's assessment is correct in that `VOLUME_INTEGRATED` parameters _should_ be adjusted when an assembly is moved to/from a position that has non-unity symmetry factor (most of the time the central assembly). I also think that he is correct in that this is not currently being done by ARMI. 

I think there are a few reasons why this has escaped notice until now:
* Not all models use 1/3 symmetry, in which case this is not relevant
* The major model that we deal with which _does_ use 1/3 symmetry also has an abnormal assembly in that position, which means that it is not easy to spot abnormal behavior based on trends in other assemblies
* Even for models in which everything is right to trigger this bug, it only has a practical impact on a small number of parameters. Most parameters are recalculated in subsequent interfaces where the symmetry factor is accurately accounted for. For those parameters, the "bug" only persists in the time between BOC and when the relevant physics interface is run. But then once the physics interface runs, the parameter is updated and eventually what is written to the database is blissfully correct. This is not the case for all parameters, though. In particular, this PR points out a number of parameters that are really only ever calculated at BOL, and they are not further updated. In these cases, those values WILL BE WRONG when they are written to the database. I believe that Brian's fix will remedy this situation.

So my assessment is that this should be fixed, and the way that Brian has proposed to fix it seems adequate to me.

I do have some small worry that other places in the code might be doing this already in a one-off sense for specific parameters. I think it would be reasonable to rely on our internal integration tests to identify any of these cases.

Just my $0.02.

Reply:
Okay, thanks Chris.

And as long as this logic all works for a non-Natrium situation (where symmetry factor will usually be `1`), I'm happy.